### PR TITLE
Add root check and install dependencies

### PR DESCRIPTION
**Pull request by pfeerick**
Add check for root permissions and install required gdisk and partprobe (part of parted) packages

### CODE REVIEW

File: package/root/usr/local/sbin/resize_rootfs.sh [hunk 1]
```diff
@@ -2,6 +2,13 @@
 
 set -xe
 
+if [ "$(id -u)" -ne "0" ]; then
+	echo "This script requires root."
+	exit 1
+fi
+
+apt-get update && apt-get install gdisk parted
```
Comment (ayufan) [inline package/root/usr/local/sbin/resize_rootfs.sh]: It is installed already.

Comment (ayufan) [inline package/root/usr/local/sbin/resize_rootfs.sh]: Or at least it should be :)

**Comment by pfeerick** [comment]
I'll try again on a fresh 0.2.3 image, but it wasn't when I tried last
night :-P


*Peter Feerick*BIT, BLDes *CQU*

File: package/root/usr/local/sbin/resize_rootfs.sh (hunk 1 above)
Comment (pfeerick) [inline package/root/usr/local/sbin/resize_rootfs.sh]: Confirmed missing on 0.2.3 image. 

Should I add them to the below lists, then I can remove the package install and leave just the root check?
https://github.com/ayufan-rock64/linux-rootfs/blob/master/configs/ubuntu/package-lists/additional.list.chroot
https://github.com/ayufan-rock64/linux-rootfs/blob/master/configs/debian/package-lists/additional.list.chroot

Comment (ayufan) [inline package/root/usr/local/sbin/resize_rootfs.sh]: Yes, please add them to these lists, but also add them here: https://github.com/ayufan-rock64/linux-build/blob/master/rootfs/make_rootfs.sh#L159.

And add them here: https://github.com/ayufan-rock64/linux-build/blob/master/Makefile#L38 as `--depends <package>`.

**Comment by pfeerick** [comment]
I *think* that's all of it now... it's up to you to test it to see if I broke anything along the way. I did test the package dependency lines as couldn't find any decent docs for fpm which explained if it accepted multiple dependencies per argument, which it looks like it doesn't.

File: Makefile [hunk 2]
```diff
@@ -30,6 +30,8 @@ linux-rock64-package-$(RELEASE_NAME).deb: package
 		--deb-compression bzip2 \
 		--after-install package/scripts/postinst.deb \
 		--before-remove package/scripts/prerm.deb \
+		--depends gdisk \
+        --depends parted \
```
Comment (ayufan) [inline Makefile]: Please make double tab.

File: rootfs/make_rootfs.sh [hunk 3]
```diff
@@ -156,7 +156,8 @@ export DEBIAN_FRONTEND=noninteractive
 locale-gen en_US.UTF-8
 apt-get -y update
 apt-get -y install dosfstools curl xz-utils iw rfkill wpasupplicant openssh-server alsa-utils \
-	nano git build-essential vim jq wget ca-certificates software-properties-common dirmngr
+	nano git build-essential vim jq wget ca-certificates software-properties-common dirmngr \
+    gdisk parted
```
Comment (ayufan) [inline rootfs/make_rootfs.sh]: Please make tab.

**Comment by ayufan** [comment]
Thanks. It looks great. Small nitpick left :)

**Comment by pfeerick** [comment]
Ah, sorry about that... I use spaces so they don't move. And then went to bed before I say saw the reply! ;)

**Pull request merged**